msal/src/test/java/com/microsoft/identity/client/robolectric/tests/AcquireTokenMockTest.java

```diff
@@ -0,0 +1,600 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.client.robolectric.tests;
+
+import android.app.Activity;
+
+import com.microsoft.identity.client.AcquireTokenParameters;
+import com.microsoft.identity.client.AcquireTokenSilentParameters;
+import com.microsoft.identity.client.AuthenticationCallback;
+import com.microsoft.identity.client.IAccount;
+import com.microsoft.identity.client.IAuthenticationResult;
+import com.microsoft.identity.client.IMultipleAccountPublicClientApplication;
+import com.microsoft.identity.client.IPublicClientApplication;
+import com.microsoft.identity.client.exception.MsalClientException;
+import com.microsoft.identity.client.exception.MsalException;
+import com.microsoft.identity.client.exception.MsalServiceException;
+import com.microsoft.identity.client.robolectric.shadows.ShadowAuthority;
+import com.microsoft.identity.client.robolectric.shadows.ShadowHttpRequest;
+import com.microsoft.identity.client.robolectric.shadows.ShadowMsalUtils;
+import com.microsoft.identity.client.robolectric.shadows.ShadowStorageHelper;
+import com.microsoft.identity.client.robolectric.shadows.ShadowStrategyResultServerError;
+import com.microsoft.identity.client.robolectric.shadows.ShadowStrategyResultUnsuccessful;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.internal.authorities.Authority;
+import com.microsoft.identity.internal.testutils.authorities.MockAuthority;
+import com.microsoft.identity.common.internal.cache.ICacheRecord;
+import com.microsoft.identity.common.internal.providers.microsoft.microsoftsts.MicrosoftStsAuthorizationRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.OAuth2Strategy;
+import com.microsoft.identity.common.internal.providers.oauth2.OAuth2TokenCache;
+import com.microsoft.identity.common.internal.providers.oauth2.TokenResponse;
+import com.microsoft.identity.internal.testutils.MockTokenResponse;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.util.Arrays;
+
+import static junit.framework.Assert.fail;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(shadows = {ShadowStorageHelper.class, ShadowAuthority.class, ShadowHttpRequest.class, ShadowMsalUtils.class})
+public final class AcquireTokenMockTest {
+
+    private static final String[] SCOPES = {"user.read"};
+    private static final String AAD_MOCK_AUTHORITY = "https://test.authority/mock";
+
+    @Test
+    public void canAcquireToken() {
```

Comment:
nit : rename something like acquireTokenSuccessTest` and all the tests below. Also a brief description of the test would be good

Reply:
Why append the word test in every test? I feel like it is unnecessary as the `@test` annotation already indicates it is a test

Reply:
It's a classic convention for readability to indicate tests,  even though we religiously didn't follow in our library in few tests, most of our tests follow this convention. 

For example, if I search for `acquireToken` in the project, i can safely ignore this method since it has `test` in it. 

Also the name here `canAcquireToken` on glance doesn't indicate to me about what we are trying to test and validate. Let's say we have 10 tests for acquire token, wouldn't this be too generic to comprehend?